api/oci/extensions/repositories/git/format.go

```diff
@@ -0,0 +1,20 @@
+package git
+
+import (
+	"ocm.software/ocm/api/oci/cpi"
+	"ocm.software/ocm/api/utils/accessobj"
+)
+
+// //////////////////////////////////////////////////////////////////////////////
+
+func Open(ctx cpi.ContextProvider, acc accessobj.AccessMode, url string, opts Options) (Repository, error) {
+	spec, err := NewRepositorySpecFromOptions(acc, url, opts)
+	if err != nil {
+		return nil, err
+	}
+	return New(cpi.FromProvider(ctx), spec, nil)
+}
+
+func Create(ctx cpi.ContextProvider, acc accessobj.AccessMode, url string, opts Options) (Repository, error) {
+	return Open(ctx, acc, url, opts)
+}
```

Comment:
No you are right, here is an inconsistency. For an arbitrary repository URL I cannot assume that we have the rights but what I am doing in Open (and currently also in Create) is creating a local repository (instead of fetching one) if it doesnt exist and then setting a remote to that branch. I am then attempting to push the repository on the next Update() call, e.g. on Close. 

Thinking about this again, we could change it so that we try to create a repo by pushing a branch with an empty "init" commit but this certainly would require rework. Im happy to hear your thoughts on the above

Reply:
Misread, enforced create/writeable